src/main/cpp/vision/DragonLimelight.cpp

```diff
@@ -188,6 +223,109 @@ std::vector<std::unique_ptr<DragonVisionStruct>> DragonLimelight::GetObjectDetec
+{
+    if (frc::RobotBase::IsSimulation())
+    {
+        return std::nullopt;
+    }
+
+    auto mode = static_cast<int>(LIMELIGHT_IMU_MODE::USE_EXTERNAL_IMU_ONLY); // Chief Delphi answer says perfect portrait pose doesn't work with internal IMU
+    LimelightHelpers::SetIMUMode(m_cameraName, mode);
+
+    if (useMegatag2)
+    {
+        return GetMegaTag2Pose();
+    }
+    else
+    {
+        return GetMegaTag1Pose();
+    }
+    return std::nullopt;
+}
+
+/// ----------------------------------------------------------------------------------
+/// @brief Get pose estimate using the MegaTag1/standard Limelight pose estimate API.
+/// @return optional VisionPose populated from Limelight pose if tagCount > 0 and valid deviations are computable.
+/// @sideeffects If robot pose has not been set, SetRobotPose will be invoked using the estimate's 2D pose.
+/// ----------------------------------------------------------------------------------
+std::optional<VisionPose> DragonLimelight::GetMegaTag1Pose()
+{
+    auto limelightMeasurement = LimelightHelpers::getBotPoseEstimate_wpiBlue(m_cameraName);
+
+    if (limelightMeasurement.tagCount == 0)
+    {
+        return std::nullopt;
+    }
+    auto deviations = GetStandardDeviationsForMetaTag1PoseEstimation(limelightMeasurement.tagCount, limelightMeasurement.avgTagArea);
+    if (!deviations.has_value())
+    {
+        return std::nullopt;
+    }
+
+    auto pose3d = frc::Pose3d{limelightMeasurement.pose};
+
+    units::time::millisecond_t currentTime = frc::Timer::GetFPGATimestamp();
+    units::time::millisecond_t timestamp = currentTime - units::millisecond_t(limelightMeasurement.timestampSeconds / 1000.0);
+
+    double xyStds = deviations.value().first;
+    double degStds = deviations.value().second;
+
+    m_megatag1PosBool = true;
+    m_megatag1Pos = {pose3d, timestamp, {xyStds, xyStds, degStds}, PoseEstimationStrategy::MEGA_TAG};
+    if (!m_robotPoseSet)
+    {
+        SetRobotPose(pose3d.ToPose2d());
+    }
+    return m_megatag1Pos;
+}
+
+/// ----------------------------------------------------------------------------------
+/// @brief Get pose estimate using the MegaTag2 specialized API.
+/// @details Requires an initial robot pose to be set (will attempt to get MegaTag1 pose first if needed).
+/// @return optional VisionPose populated from MegaTag2 pose estimate API; std::nullopt on failure.
+/// ----------------------------------------------------------------------------------
+std::optional<VisionPose> DragonLimelight::GetMegaTag2Pose()
+{
+    if (!m_robotPoseSet)
+    {
+        auto megatag1pose = GetMegaTag1Pose();
+        if (!megatag1pose.has_value())
+        {
+            return std::nullopt;
+        }
+    }
+    // Get the pose estimate
+    auto mode = frc::DriverStation::IsDisabled() ? static_cast<int>(LIMELIGHT_IMU_MODE::USE_EXTERNAL_IMU_AND_FUSE_WITH_INTERNAL_IMU) : static_cast<int>(LIMELIGHT_IMU_MODE::USE_INTERNAL_IMU);
```

Comment:
Why are we using the fused IMU data? If we are disabled, shouldn't we just use the IMU that the limelight sets based on tags?

Reply:
This is what we currently do.  I think it is ok because we keep refreshing until the pose gets set from the tags.